### PR TITLE
add temperature monitoring feature to heartbeat.js

### PR DESCRIPTION
**Pull request by jimyang2008**

### CODE REVIEW

File: cli/heartbeat.js [hunk 1]
```diff
@@ -211,10 +212,96 @@ async function update(status, extra) {
   return info;
 }
 
+async function getOverheatedThresholds() {
+  const uname = await getShellOutput("uname -m");
+  let temperatureThreshold = null;
+  let timeThreshold = null;
+  switch (uname) {
+    case "aarch64": {
+      const boardName = await getShellOutput("awk -F= '/BOARD=/ {print $2}' /etc/firewalla-release");
```
Comment (MelvinTo) [inline cli/heartbeat.js:221]: cache the local threshold per process life time, no need to query per interval

Comment (jimyang2008) [reply cli/heartbeat.js:221]: Correct me if I am wrong about `setInterval` call. 

The thresholds are calculated inside `getOverheatedThresholds()`, which is only invoked once inside `setInterval()` call. And later run of `monitorTemperature` will just use the same passed-in value, I suppose.

Comment (jimyang2008) [reply cli/heartbeat.js:226]: Go with the way of global variable

File: cli/heartbeat.js [hunk 2]
```diff
@@ -211,10 +218,97 @@ async function update(status, extra) {
...
+  }
+  return ;
+}
+
 const job = setInterval(() => {
   update("schedule");
 }, 24 * 3600 * 1000); // every day
 
+const jobTemperature = setInterval(async () => {
+  await monitorTemperature(false);
```
Comment (MelvinTo) [inline cli/heartbeat.js:309]: remove false here

File: cli/heartbeat.js [hunk 3]
```diff
@@ -211,10 +218,97 @@ async function update(status, extra) {
...
+}, 3 * 1000); // every 30 seconds
```
Comment (MelvinTo) [inline cli/heartbeat.js:310]: 3 => 30

File: cli/heartbeat.js [hunk 4]
```diff
@@ -52,6 +53,12 @@ const launchTime = Math.floor(new Date() / 1000);
 
 let uid = null;
 
+let sysStateCount = { "normal": 0, "overheated": 0 };
+let overheatedThresholds = null;
+(async function() {
```
Comment (MelvinTo) [inline cli/heartbeat.js:58]: minor: run setInterval at the end of this function instead of main-loop

Comment (jimyang2008) [reply cli/heartbeat.js:58]: Fixed as suggested

File: cli/heartbeat.js (hunk 3 above)
Comment (jimyang2008) [reply cli/heartbeat.js]: fixed

File: cli/heartbeat.js (hunk 2 above)
Comment (jimyang2008) [reply cli/heartbeat.js]: fixed

**Pull request merged**